packages/flutter/test/painting/image_provider_and_image_cache_test.dart

```diff
@@ -46,40 +44,40 @@ void main() {
 
     const ImageProvider provider = ExactAssetImage('does-not-exist');
     final Object key = await provider.obtainKey(ImageConfiguration.empty);
-    expect(imageCache.statusForKey(provider).untracked, true);
-    expect(imageCache.pendingImageCount, 0);
+    expect(imageCache!.statusForKey(provider).untracked, true);
+    expect(imageCache!.pendingImageCount, 0);
 
     provider.resolve(ImageConfiguration.empty);
 
-    expect(imageCache.statusForKey(key).pending, true);
-    expect(imageCache.pendingImageCount, 1);
+    expect(imageCache!.statusForKey(key).pending, true);
+    expect(imageCache!.pendingImageCount, 1);
 
     await error.future;
 
-    expect(imageCache.statusForKey(provider).untracked, true);
-    expect(imageCache.pendingImageCount, 0);
+    expect(imageCache!.statusForKey(provider).untracked, true);
+    expect(imageCache!.pendingImageCount, 0);
   }, skip: isBrowser); // https://github.com/flutter/flutter/issues/56314
 
   test('AssetImageProvider - evicts on null load', () async {
-    final Completer<StateError> error = Completer<StateError>();
+    final Completer<Object> error = Completer<Object>();
     FlutterError.onError = (FlutterErrorDetails details) {
-      error.complete(details.exception as StateError);
+      error.complete(details.exception);
     };
 
     final ImageProvider provider = ExactAssetImage('does-not-exist', bundle: _TestAssetBundle());
     final Object key = await provider.obtainKey(ImageConfiguration.empty);
-    expect(imageCache.statusForKey(provider).untracked, true);
-    expect(imageCache.pendingImageCount, 0);
+    expect(imageCache!.statusForKey(provider).untracked, true);
+    expect(imageCache!.pendingImageCount, 0);
 
     provider.resolve(ImageConfiguration.empty);
 
-    expect(imageCache.statusForKey(key).pending, true);
-    expect(imageCache.pendingImageCount, 1);
+    expect(imageCache!.statusForKey(key).pending, true);
+    expect(imageCache!.pendingImageCount, 1);
 
     await error.future;
 
-    expect(imageCache.statusForKey(provider).untracked, true);
-    expect(imageCache.pendingImageCount, 0);
+    expect(imageCache!.statusForKey(provider).untracked, true);
+    expect(imageCache!.pendingImageCount, 0);
```

Comment:
This expect is failing:

```
package:test_api                                             expect
package:flutter_test/src/widget_tester.dart 429:3            expect
test/painting/image_provider_and_image_cache_test.dart 80:5  main.<fn>

Expected: <0>
  Actual: <1>
```

I doesn't make sense to me why migrating this to NNBD would cause that... /cc @dnfield any idea?

Reply:
Why are we changing the completer from `Completer<StateError>` to `Completer<void>`?I don't see how that would cause this problem but it seems strange to me.

Reply:
I think I got it, this test case was supposed to test this error case here:

https://github.com/flutter/flutter/blob/c8466d0430195755615a987171920582823ed79a/packages/flutter/lib/src/painting/image_provider.dart#L676-L682

However, as the comment there says, with null-safety enabled, that error cannot happen anymore. Therefore deleting this test case.